### ISSUE TITLE
conversion from float16 to bfloat16 generates different results between CPU and Spyre

### ISSUE THREAD

**Issue by kiszk** [torch-spyre/torch-spyre#2478]
### What happened?

When we execute type conversion of a tensor from float16 to bfloat16, the results are different between CPU and Spyre.
Tensor on CPU device seems to do in nearest mode. On the other hand, tensor on Spyre device seems to do in toward zero.

Maybe it is related to https://github.com/torch-spyre/torch-spyre/issues/2343

### What did you expect to happen?

The conversion results should be the same (`18.125`) instead of `18.0`

Float16 on Spyre is handled as dlfloat16. Dfload16 can also represent a given value ('18.09375') without loss. So, we expect that these two type conversions should return the same value.

### How can we reproduce it?

repro
```python
import torch

torch.set_printoptions(precision=7)

def test(x):
    return x.to(dtype=torch.bfloat16)

#                    sign, exponent, mantissa
# float16  (1,5,10) [   0, 10011,    0010000110] = 18.09375
# dlfloat16(1,6, 9) [   0, 100011,   001000011 ] = 18.09375
# bfloat16 (1,8, 7) [   0, 10000011, 0010001   ] = 18.125
# bfloat16 (1,8, 7) [   0, 10000011, 0010000   ] = 18.000

x = torch.full((2,), 18.09375, dtype=torch.float16).cpu()
compiled = torch.compile(test, backend="inductor")
z = compiled(x)
print("float16 -> bfloat16 on cpu  :", x, z)

x = torch.full((2,), 18.09375, dtype=torch.float16).to("spyre")
compiled = torch.compile(test, backend="inductor")
z = compiled(x)
print("float16 -> bfloat16 on Spyre:", x, z)
```

results
```
float16 -> bfloat16 on cpu  : tensor([18.0937500, 18.0937500], dtype=torch.float16) tensor([18.1250000, 18.1250000], dtype=torch.bfloat16)
float16 -> bfloat16 on Spyre: tensor([18.0937500, 18.0937500], dtype=torch.float16, device='spyre:0') tensor([18., 18.], dtype=torch.bfloat16, device='spyre:0')
```

### Any environmental details we need to know?

torch-spyre commit : https://github.com/torch-spyre/torch-spyre/commit/95eaf1ff4a5a32abf802d6d160372c1d8b7286a8

### Anything else we need to know?

sdsc.json
```
{
  "0_identity": {
    "sdscFoldProps_": [
      {
        "factor_": 1,
        "label_": "time"
      }
    ],
    "sdscFolds_": {
      "dim_prop_func": [
        {
          "Affine": {
            "alpha_": 1,
            "beta_": 0
          }
        }
      ],
      "dim_prop_attr": [
        {
          "factor_": 1,
          "label_": "time"
        }
      ],
      "data_": {
        "[0]": "0"
      }
    },
    "coreFoldProp_": {
      "factor_": 1,
      "label_": "core"
    },
    "coreletFoldProp_": {
      "factor_": 1,
      "label_": "corelet"
    },
    "numCoresUsed_": 1,
    "coreIdToDsc_": {
      "0": 0
    },
    "numWkSlicesPerDim_": {
      "out": 1
    },
    "coreIdToWkSlice_": {
      "0": {
        "out": 0
      }
    },
    "coreIdToDscSchedule": {
      "0": [
        [
          -1,
          0,
          0,
          0
        ]
      ]
    },
    "dscs_": [
      {
        "identity": {
          "numCoresUsed_": 1,
          "numCoreletsUsed_": 1,
          "coreIdsUsed_": [
            0
          ],
          "N_": {
            "name_": "n",
            "out_": 64
          },
          "coordinateMasking_": {},
          "maskingConstId_": -1,
          "dataStageParam_": {
            "0": {
              "ss_": {
                "name_": "core",
                "out_": 64
              },
              "el_": {
                "name_": "core",
                "out_": 64
              }
            }
          },
          "primaryDsInfo_": {
            "OUTPUT": {
              "layoutDimOrder_": [
                "out"
              ],
              "stickDimOrder_": [
                "out"
              ],
              "stickSize_": [
                64
              ]
            }
          },
          "scheduleTree_": [
            {
              "nodeType_": "allocate",
              "name_": "allocate-Tensor0_hbm",
              "prev_": "",
              "ldsIdx_": 0,
              "component_": "hbm",
              "layoutDimOrder_": [
                "out"
              ],
              "maxDimSizes_": [
                -1
              ],
              "startAddressCoreCorelet_": {
                "dim_prop_func": [
                  {
                    "Map": {}
                  },
                  {
                    "Const": {}
                  },
                  {
                    "Const": {}
                  }
                ],
                "dim_prop_attr": [
                  {
                    "factor_": 1,
                    "label_": "core"
                  },
                  {
                    "factor_": 1,
                    "label_": "corelet"
                  },
                  {
                    "factor_": 1,
                    "label_": "time"
                  }
                ],
                "data_": {
                  "[0, 0, 0]": "0"
                }
              },
              "coordinates_": {
                "coordInfo": {
                  "out": {
                    "spatial": 3,
                    "temporal": 0,
                    "elemArr": 2,
                    "padding": "nopad",
                    "folds": {
                      "dim_prop_func": [
                        {
                          "Affine": {
                            "alpha_": 64,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 0,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 0,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 64,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 1,
                            "beta_": 0
                          }
                        }
                      ],
                      "dim_prop_attr": [
                        {
                          "factor_": 1,
                          "label_": "core_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "corelet_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "row_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "elem_arr_1"
                        },
                        {
                          "factor_": 64,
                          "label_": "elem_arr_0"
                        }
                      ]
                    }
                  }
                },
                "coreIdToWkSlice_": {}
              }
            },
            {
              "nodeType_": "allocate",
              "name_": "allocate-Tensor1_hbm",
              "prev_": "",
              "ldsIdx_": 1,
              "component_": "hbm",
              "layoutDimOrder_": [
                "out"
              ],
              "maxDimSizes_": [
                -1
              ],
              "startAddressCoreCorelet_": {
                "dim_prop_func": [
                  {
                    "Map": {}
                  },
                  {
                    "Const": {}
                  },
                  {
                    "Const": {}
                  }
                ],
                "dim_prop_attr": [
                  {
                    "factor_": 1,
                    "label_": "core"
                  },
                  {
                    "factor_": 1,
                    "label_": "corelet"
                  },
                  {
                    "factor_": 1,
                    "label_": "time"
                  }
                ],
                "data_": {
                  "[0, 0, 0]": "17179869184"
                }
              },
              "coordinates_": {
                "coordInfo": {
                  "out": {
                    "spatial": 3,
                    "temporal": 0,
                    "elemArr": 2,
                    "padding": "nopad",
                    "folds": {
                      "dim_prop_func": [
                        {
                          "Affine": {
                            "alpha_": 64,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 0,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 0,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 64,
                            "beta_": 0
                          }
                        },
                        {
                          "Affine": {
                            "alpha_": 1,
                            "beta_": 0
                          }
                        }
                      ],
                      "dim_prop_attr": [
                        {
                          "factor_": 1,
                          "label_": "core_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "corelet_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "row_fold"
                        },
                        {
                          "factor_": 1,
                          "label_": "elem_arr_1"
                        },
                        {
                          "factor_": 64,
                          "label_": "elem_arr_0"
                        }
                      ]
                    }
                  }
                },
                "coreIdToWkSlice_": {}
              }
            }
          ],
          "labeledDs_": [
            {
              "ldsIdx_": 0,
              "dsName_": "Tensor0",
              "dsType_": "OUTPUT",
              "scale_": [
                1
              ],
              "wordLength": 2,
              "dataFormat_": "SEN169_FP16",
              "memOrg_": {
                "hbm": {
                  "isPresent": 1
                },
                "lx": {
                  "isPresent": 1
                }
              }
            },
            {
              "ldsIdx_": 1,
              "dsName_": "Tensor1",
              "dsType_": "OUTPUT",
              "scale_": [
                1
              ],
              "wordLength": 2,
              "dataFormat_": "SEN169_FP16",
              "memOrg_": {
                "hbm": {
                  "isPresent": 1
                },
                "lx": {
                  "isPresent": 1
                }
              }
            }
          ],
          "constantInfo_": "{}",
          "computeOp_": [
            {
              "exUnit": "sfp",
              "opFuncName": "identity",
              "attributes_": {
                "dataFormat_": "SEN169_FP16",
                "fidelity_": "regular"
              },
              "location": "Inner",
              "inputLabeledDs": [
                "Tensor0-idx0"
              ],
              "outputLabeledDs": [
                "Tensor1-idx1"
              ]
            }
          ]
        }
      }
    ]
  }
}
```

### Relevant log output

```Shell

```

**Comment by kiszk** [comment]
@manid2 Could you take a look?

**Comment by jturney** [comment]
I have a possible fix for this in #2524

**Comment by kiszk** [comment]
**DeepTools #4338** has been merged, and I confirmed that this is solved.
